src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSize.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSize : SmokeTests
+{
+    private readonly string baselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private readonly string[] baselineFileContent;
+    private readonly Regex buildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+
+
+    public ArtifactsSize(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        baselineFileContent = File.Exists(baselineFilePath) ? File.ReadAllLines(baselineFilePath) : Array.Empty<string>();
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpace: true)]
+    public void ArtifactsSizeTest()
+    {
+        Assert.True(Directory.Exists(BaselineHelper.GetBaselineFilePath("ArtifactsSizes/")));
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+
+        string[] tarEntries = sdkTarEntries.Concat(artifactsTarEntries)
+            .Select(entry =>
+            {
+                string modifiedPath = buildVersionPattern.Replace(entry.Name, "VERSION");
+                string result = modifiedPath.Replace(Config.TargetRid, "TARGET_RID");
+                return $"{result}: {entry.Length} bytes";
```

Comment:
It doesn't seem necessary to include `bytes` in the output files.  It feels like unnecessary information that bloats the file and requires special handling to parse.  Bytes can be inferred by the file structure convention.